Define `profiling.instrumentation.source` Log Message Attribute (#336)
signalfx/gdi-specification: specification/semantic_conventions.md

  - `pprof-gzip-base64`,
  - `text` ([Deprecated](../README.md#versioning-and-status-of-the-specification)
    format).

> - `profiling.instrumentation.source` MUST be set to either `continuous` or `snapshot` to 
>   identify the profiling mechanism reporting the callstacks
>   - `continuous` SHOULD be used to indicate the continuous "Always On" profiling
>   - `snapshot` SHOULD be used to indicate the trace snapshot profiling

Reviewer: how about

```suggestion
- `profiling.instrumentation.source` MUST be set to either:
  - `continuous` for continuous "Always On" profiling
  - `snapshot` for trace snapshot profiling
```

  - `pprof-gzip-base64`,
  - `text` ([Deprecated](../README.md#versioning-and-status-of-the-specification)
    format).

> - `profiling.instrumentation.source` MUST be set to either `continuous` or `snapshot`

Reviewer 2: I think we should add a few more words here that describe _when_ one or the other should be used and briefly what they mean. Like literally one or two sentences.

Author: Additional context added. Feedback is very welcome!

  - `pprof-gzip-base64`,
  - `text` ([Deprecated](../README.md#versioning-and-status-of-the-specification)
    format).
- `profiling.instrumentation.source` MUST be set to either:
  - `continuous` for continuous "Always On" profiling

>   - `snapshot` for trace snapshot profiling

Reviewer 3: I think that, for backward compatibility, you should mention that it will be treated as `continuous`.

Reviewer 4: This, there are too many existing installations, my suggestion is that `profiling.instrumentation.source` must only be set when anything other than `continuous` is used (as the backend treats all of the profiles as continuous by default).
